src/neptune/common/warnings.py

```diff
@@ -55,16 +56,19 @@ def get_user_code_stack_level():
 
 
 def warn_once(message: str, *, exception: type(Exception) = None):
-    if exception is None:
-        exception = NeptuneDeprecationWarning
-
-    if message not in warned_once:
-        warnings.warn(
-            message=message,
-            category=exception,
-            stacklevel=get_user_code_stack_level(),
-        )
-        warned_once.add(message)
+    if len(warned_once) < 1_000:
```

Comment:
I think we should change this behavior to actually clean up the set, because some important messages might be lost. A simple LRU queue or even bruteforce removal of a constant number of elements

Reply:
I don't know. If something goes really wrong do we actually want an infinite stream of errors?

Reply:
> some important messages 
These messages will be logged by different loggers.